File: android/DFM/src/main/java/com/dontflymoney/api/InternalResponse.java
package com.dontflymoney.api;

import org.json.JSONObject;

class InternalResponse
{
	private JSONObject result;
	private String error;

	InternalResponse(JSONObject result)
	{
		this.result = result;
	}

	InternalResponse(String error)
	{
		this.error = error;
	}
	

	boolean IsSuccess()
	{
		return error == null;
	}
	
	JSONObject GetResult()
	{
		return result;
	}
	
	String GetError()
	{
		return error;
	}
	
	
}

File: android/DFM/src/main/java/com/dontflymoney/api/InternalResponse.kt
package com.dontflymoney.api

import org.json.JSONObject

internal class InternalResponse {
    private val result: JSONObject
    private val error: String?

    constructor(result: JSONObject) {
        this.result = result
    }

    constructor(error: String) {
        this.error = error
    }


    fun IsSuccess(): Boolean {
        return error == null
    }

    fun GetResult(): JSONObject {
        return result
    }

    fun GetError(): String {
        return error
    }


}
